Examples/StarknetDemo/StarknetDemo/AccountsStore.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  AccountsStore.swift
+//  StarknetDemo
+//
+//  Created by Bartosz Rybarski on 20/01/2023.
+//
+
+import Foundation
+import Starknet
+import BigInt
+
+let erc20: Felt = "0x49D36570D4E46F48E99674BD3FCC84644DDD6B96F7C741B1562B82F9E004DC7"
+
+func makeAccount(provider: StarknetProviderProtocol, privateKey: Felt, address: Felt) -> StarknetAccountProtocol {
+    let signer = StarkCurveSigner(privateKey: privateKey)!
+    
+    let account = StarknetAccount(
+        address: address,
+        signer: signer,
+        provider: provider
+    )
+    
+    return account
+}
+
+class AccountsStore: ObservableObject {
+    let accounts: [StarknetAccountProtocol]
+    let provider: StarknetProviderProtocol
+    
+    @Published var currentAccountIndex = 0 {
+        didSet {
+            Task {
+                await fetchBalance()
+            }
+        }
+    }
+    @Published var accountBalances: [BigUInt]
+    
+    @Published var loading = false
+    
+    var account: StarknetAccountProtocol {
+        return accounts[currentAccountIndex]
+    }
+    
+    var balance: BigUInt {
+        return accountBalances[currentAccountIndex]
+    }
+    
+    init() {
+        let provider = StarknetProvider(starknetChainId: .testnet, url: "http://localhost:5050/rpc")!
+        self.provider = provider
+        
+        let account1 = makeAccount(
+            provider: provider,
+            privateKey: "0xe3e70682c2094cac629f6fbed82c07cd",
+            address: "0x7e00d496e324876bbc8531f2d9a82bf154d1a04a50218ee74cdd372f75a551a"
+        )
+        
+        let account2 = makeAccount(
+            provider: provider,
+            privateKey: "0xf728b4fa42485e3a0a5d2f346baa9455",
+            address: "0x69b49c2cc8b16e80e86bfc5b0614a59aa8c9b601569c7b80dde04d3f3151b79"
```

Comment:
Perhaps we could move the variables at the top, so everything is defined in one place? I guess it could be easier for someone to modify them and play around with the demo.

Reply:
We could maybe use some inline comments explaining what is happening? demo app by default could be a nice source of knowledge for someone completely new to starknet then

Reply:
Absolutely, good idea!